extensions/dacpac/src/wizard/dataTierApplicationWizard.ts

```diff
@@ -288,34 +314,82 @@ export class DataTierApplicationWizard {
 	}
 
 	public async deploy(): Promise<mssql.DacFxResult> {
+		const deployStartTime = new Date().getTime();
 		const service = await this.getService(msSqlProvider);
 		const ownerUri = await azdata.connection.getUriForConnection(this.model.server.connectionId);
 
-		return await service.deployDacpac(this.model.filePath, this.model.database, this.model.upgradeExisting, ownerUri, azdata.TaskExecutionMode.execute);
+		const result = await service.deployDacpac(this.model.filePath, this.model.database, this.model.upgradeExisting, ownerUri, azdata.TaskExecutionMode.execute);
+
+		// Deploy Dacpac: 'Deploy button' clicked in deploy summary page, Reporting the event selection to the telemetry
```

Comment:
For all these - are you also capturing failures so you can get an accurate count of total deployments/extractions/etc, not just the successful ones?

(based on the code I'm guessing that it'll return with a failure message if an error occurs in the service but you should also be capturing other errors that can happen before it gets to that point such as JSON RPC errors or something)

Reply:
Working on this one too!!

Reply:
And just to be clear - one thing I'm suggesting here is that you surround the deployDacpac in a try/catch and then if an error is thrown you send an event with the fields filled in from that.

Reply:
@Charles-Gagnon, In the catch block, exception error message might have PII information(would they? or gives a generic message). So I was thinking by adding an telemetry event which gives isServiceExists, isDatabaseNameNull, isFilePathNull? kind of information. Here, result will be null only if the service is unavailable that can be captured by catch block, But if any of the dac service arguments are null, the result will not be null and won't throw exception, so for that I have to use another null check and send a telemetry event. Any suggestions here?

Reply:
Yes, you shouldn't send the error message from the exception directly. I would say just logging that an error occurred is a good start since this is unlikely to ever happen anyways.

Is the result being undefined actually a valid scenario? I would expect it to either return a result or throw an exception.

Reply:
Yes since the result always return DacFxResult properties such as success and error updated, it will return those fields if any error occurred. Will add extra null check to include the props and send it to PR, we can check that one and update if need any change :)